extensions/omniv21/fileformat/edi/seg.go

```diff
@@ -0,0 +1,127 @@
+package edi
+
+import (
+	"github.com/jf-tech/go-corelib/maths"
+	"github.com/jf-tech/go-corelib/strs"
+)
+
+const (
+	segTypeSeg   = "segment"
+	segTypeGroup = "segment_group"
+)
+
+const (
+	fqdnDelim = "/"
+)
+
+type elem struct {
+	Name           string `json:"name,omitempty"`
+	Index          int    `json:"index,omitempty"`
+	ComponentIndex *int   `json:"component_index,omitempty"`
+	EmptyIfMissing bool   `json:"empty_if_missing,omitempty"`
+}
+
+type segDecl struct {
+	Name     string     `json:"name,omitempty"`
+	Type     *string    `json:"type,omitempty"`
+	IsTarget bool       `json:"is_target,omitempty"`
+	Min      *int       `json:"min,omitempty"`
+	Max      *int       `json:"max,omitempty"`
+	Elems    []elem     `json:"elements,omitempty"`
+	Children []*segDecl `json:"child_segments,omitempty"`
+}
+
+func (d *segDecl) isGroup() bool {
+	return d.Type != nil && *d.Type == segTypeGroup
+}
+
+func (d *segDecl) minOccurs() int {
+	switch d.Min {
+	case nil:
+		return 1
```

Comment:
Why we treat nil as min as 1 instead of 0?

Reply:
Maybe because of simply the general situation that element occur exact once then we don't need to specify? Not sure whether we should add some comments here.

Reply:
Yes that's the intent. Vast majority seg is min:1 and max:1. Will add comment